scripts/wad-diff

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+# wad-diff - Generate a diff image for each WAD file that has changed.
+# by Steven Elliott <selliott512@gmail.com>
+#
+# The set of changed WAD files and the changes included in them is driven by
+# the optional "commit" argument. The commit argument is similar to commit
+# argument that may be passed to "git diff". Like with "git diff" by default
+# the comparison is from HEAD to the workspace. If a single revision commit
+# argument (no "..") is specified then that becomes the from revision instead
+# of HEAD. If a two revision commit argument (with a "..") is specified then
+# the difference is between those two revisions, making the workspace
+# irrelevant.
+#
+# The set of WAD files considered can be further reduced by the optional
+# "wad-regex" argument. For example, if a diff would be produced for both map05
+# and map07 but you only want to see the diff for map05 you could pass "05" for
+# wad-regex.
+
+# Globals
+
+dname="${0%/*}"
+bname="${0##*/}"
+tmp_dir="/tmp/$bname.$$"
```

Comment:
I'm aware of `mktemp`, but I use the approach above because the I like the having the PID in the name. If the PID is logged somewhere, or seen from `ps`, it makes it easy to relate the running process to it's temporary files. However, there's something to be said for following standard secure programming practices so that others can make sure the program is secure even when you've convinced yourself that it's secure. So for this script, which will probably be changed a bit, renamed, and moved to `wad2image`, but will still exist, I'll do as you suggest. But I may revert back to my `tmp_dir` ways in future scripts I write. I'll see how it goes :-)